ground/src/main/java/com/google/android/ground/repository/SurveyRepository.kt

```diff
@@ -106,15 +107,11 @@ constructor(
    */
   suspend fun loadAndSyncSurveyWithRemote(id: String): Survey? {
     Timber.d("Loading survey $id")
-    val survey =
-      withTimeout(LOAD_REMOTE_SURVEY_TIMEOUT_MILLS) { remoteDataStore.loadSurvey(id) }
-        ?: return null
-
-    localSurveyStore.insertOrUpdateSurvey(survey)
-    survey.tileSources.forEach {
-      tileSourceDao.insertOrUpdateSuspend(it.toLocalDataStoreObject(survey.id))
-    }
-    return survey
+    return withTimeoutOrNull(LOAD_REMOTE_SURVEY_TIMEOUT_MILLS) { remoteDataStore.loadSurvey(id) }
+      ?.apply {
+        localSurveyStore.insertOrUpdateSurvey(this)
```

Comment:
Can we write this as sequential operations to decrease nesting here?

eg 
```kotlin
val survey = await withTimeoutOrNull(LOAD_REMOTE_SURVEY_TIMEOUT_MILLS) { remoteDataStore.loadSurvey(id) } ?: return null

localSurveyStore.insertOrUpdateSurvey(this)
tileSources.forEach { tileSourceDao.insertOrUpdateSuspend(it.toLocalDataStoreObject(id)) }
```

Also, one silly question - what happens here if you don't `await` on `insertOrUpdateSurvey` and `insertOrUpdateSuspend`? Are Coroutines smart enough to run them in separate threads, and the wait for all the complete before returning from this suspend function? Or does something else happen?

Reply:
Thanks for the suggestion! SG to rewrite as a sequential operation.

> Also, one silly question - what happens here if you don't await on insertOrUpdateSurvey and insertOrUpdateSuspend? Are Coroutines smart enough to run them in separate threads, and the wait for all the complete before returning from this suspend function? Or does something else happen?

`await` is required only when we are using a Rx object inside a coroutine. When calling a `suspend` function from within a `suspend` function, it doesn't switch threads if we haven't explicitly mentioned the dispatcher.